core/boot/src/main/java/org/eclipse/dataspaceconnector/boot/system/DefaultServiceExtensionContext.java

```diff
@@ -281,7 +283,12 @@ private void addDefaultExtensions(List<ServiceExtension> loadedExtensions) {
 
         var injectFields = Arrays.stream(ext.getClass().getDeclaredFields())
                 .filter(f -> f.getAnnotation(Inject.class) != null)
-                .map(f -> new FieldInjectionPoint<>(ext, f, getFeatureValue(f.getType())));
+                .map(f -> {
+                    var isRequired = f.getAnnotation(Inject.class).required();
+                    var fip = new FieldInjectionPoint<>(ext, f, getFeatureValue(f.getType()));
```

Comment:
This method is equal to `InjectPointScanner.getInjectionPoints`, so the duplication could be removed making this method calling the `InjectPointScanner`.

Reply:
good catch!